package.json

```diff
@@ -1666,6 +1806,16 @@
           "type": "boolean",
           "default": true,
           "description": "%cmake-tools.configuration.cmake.exportCompileCommandsFile.description%"
+        },
+        "cmake.useCMakePresets":  {
+          "type": "string",
+          "enum": [
+            "true",
```

Comment:
I think it looks a little weird in the settings.json to have these booleans be strings.  I have the setting right next to another boolean setting and one has quotes while the other does not. Is there a better way?

Reply:
We can use any strings with similar meanings such as 'yes' and 'no'. true/false/undefined is a harder choice since config.ts doesn't distinguish the source of the settings, so we need to read config only for this setting, which makes maintenance harder.

Reply:
I agree.  We have a bunch of settings that are type: boolean with default: null and vs code kicks the user back to json because it can't be edited in the UI.  If we used "oneOf" with boolean or enum: auto, would that help?

Reply:
Yeah this is a good idea.

Reply:
It seems the oneOf approach requires editing the setting in JSON, but I opened a feature request on the VS Code team to evaluate whether we can get tri-state bools somehow.  https://github.com/microsoft/vscode/issues/120991